redash/query_runner/pg.py

```diff
@@ -121,29 +121,14 @@ def run_query(self, query):
             cursor.execute(query)
             _wait(connection)
 
-            # While set would be more efficient here, it sorts the data which is not what we want, but due to the small
-            # size of the data we can assume it's ok.
-            column_names = []
-            columns = []
-            duplicates_counter = 1
-
             if cursor.description is not None:
-                for column in cursor.description:
-                    # TODO: this deduplication needs to be generalized and reused in all query runners.
-                    column_name = column.name
-                    if column_name in column_names:
-                        column_name += str(duplicates_counter)
-                        duplicates_counter += 1
-
-                    column_names.append(column_name)
+                columns_data = self.fetch_columns([(i.name, types_map.get(i.type_code, None)) for i in cursor.description])
 
-                    columns.append({
-                        'name': column_name,
-                        'friendly_name': column_name,
-                        'type': types_map.get(column.type_code, None)
-                    })
+                columns = [{'name': col[0],
+                            'friendly_name': col[0],
+                            'type': col[1]} for col in columns_data]
```

Comment:
Maybe change `fetch_columns` method to return an array in this format? It will save the need to make another loop at the array.